Antag reputation port

ike709 opened the pull request:
:cl: avosirenfal, ported by ike709
add: Every round you finish alive and active, your chances of rolling antag increases until the odds are double and then it resets once you roll antag.
add: Certain jobs buff your antag chances more than others, usually based on the job's importance. AKA don't play assistant if you want to frequently be an antag.
/:cl:

Ports https://github.com/tgstation/tgstation/pull/35485/

Values for each job are different from TG and subject to maintainer review:
```
AI - 10
Captain - 20
Chief Engineer - 18
Chief Medical Officer - 18
Detective - 10
Head of Personnel - 18
Head of Security - 18
Research Director - 18
Security Officer - 15
Warden - 16
Atmospheric Technician - 12
Botanist - 8
Chemist - 12
Cook - 8
Cyborg - 8
Geneticist - 12
Janitor - 8
Internal Affairs Agent - 12
Medical Doctor - 12
Quartermaster - 12
Roboticist - 12
Scientist - 12
Shaft Miner - 10
Station Engineer - 12
Virologist - 12
Bartender - 8
Cargo Technician - 10
Chaplain - 5
Clown - 4
Curator - 5
Mime - 4
Blueshield - 14
Paramedic - 12
Brig Physician - 10
Assistant - 3
```

I'm accepting suggestions for ways of displaying the values to players without breaking immersion. Just shove it on the wiki? It'd be lame, but easy.

EDIT:
>I refactored how the code works and also added some job checks. Here's how it works now, checked at round end:
>Roll antag? Points removed.
Don't survive the round or go AFK? Zero points.
Get an ID change from lower point to higher point job? You get your original job points.
Get an ID change from higher point job to lower point job? You get the lower point job points. That's how being demoted or being a prisoner at roundend handles this.

~~DNM pending a fix port from TG.~~ Ported #36188

ike709 commented:
Also note that this is disabled by default in the config and will need to be enabled.

ike709 commented:
Currently thinking about buffing IAA to 14 but otherwise I'm pretty happy with the values.

AdamElTablawy commented:
Man, I sure love tons and tons of fucking traitor AIs over and over again, and dying uncontrollably to things I can't prevent fucking me over my chances of getting antag.

ike709 commented:
Yeah it's pretty great.

AdamElTablawy commented:
i can't remember why i was against this give me free antag

FlattestGuitar commented:
Adam actually has a good point. The vast majority of AI players pick the job *most* of the time. This would mean that with this PR we're looking at the AI antag frequency almost doubling.

Cactus2388 commented:
I feel like Shaft Miner should give more than Cargo Tech

SamHPurp commented:
Dislike, I much prefer the dice roll, rather than any guiding hand to help increase or decrease these chances - as ultimately they will be gamed.

Cactus2388 commented:
It'll be gamed but we'll have people helpful roles. No more 6 assistants and 0 sec.

Quantum-M commented:
But will said people be helpful in said roles? And what if they get demoted?

Ziiro commented:
I think rewarding people for actually playing roles that aren't assistant is important, and I think that this accomplishes that quite nicely.

ike709 commented:
@FlattestGuitar suggest a new value for AI then

ike709 commented:
@Quantum-M If someone is doing a shit job as a role, they should be jobbanned. Problem solved, as this should already be the case with shitty players.

Ziiro commented:
AI and borg should be halved, imo

ike709 commented:
yeah sure fuck it

ike709 commented:
AI and borg changed to 10 and 8 respectively.

SmokingKilz commented:
I agree with Ziiro. Encouraging more players to play helpful roles as opposed to assistant will go a long way to have more sec, command, etc, players.

Cactus2388 commented:
Is it round-start jobs only? What stops me from changing my ID and access to captain and grind the system?

ike709 commented:
@Cactus2388 It is your starting job yes. What stops you is the fact that it assigns your points when your job is assigned.

Cactus2388 commented:
So I can start as higher ranked job and icly get a job change to grind points

ike709 commented:
@Cactus2388 not anymore

ike709 commented:
I refactored how the code works and also added some job checks. Here's how it works now, checked at round end:

Roll antag? Points removed.
Don't survive the round or go AFK? Zero points.
Get an ID change from lower point to higher point job? You get your original job.
Get an ID change from higher point job to lower point job? You get the lower point job. That's how being demoted or being a prisoner at roundend handles this.

ike709 commented:
I did some very basic testing after the refactor but further testing is needed.

Ziiro commented:
I'd like to point out that this makes doing your job, surviving, and not getting demoted actually important.

SmokingKilz commented:
You know, I'd actually mention how that could bring up an issue of people playing SUPER safe but... I'm fine with that. Biggest reason murderbone is so prevalent on any server (apart from shitters) is that death is really not REALLY that big of a danger. This gives incentive to play safer, I like it.

ike709 commented:
@Ziiro *a pretty big _good_ change

ike709 commented:
So far the only complaints I've received after the latest batch of changes (other than point values), are all related to things that should already be ahelped anyways. So I'd say this just further incentivizes people to ahelp shitty behavior that should have been ahelped anyways.

Ziiro commented:
>@Ziiro *a pretty big good change

That's actually what I meant. I like this a lot.

ike709 commented:
Also notice how I said round end not server restart. So people can still have their free for all on centcom.

AndrewMontagne commented on code/controllers/subsystem/persistence.dm:
```diff
@@ -197,3 +215,48 @@ SUBSYSTEM_DEF(persistence)
+	var/ANTAG_REP_MAXIMUM = CONFIG_GET(number/antag_rep_maximum)
+
+	for(var/p_ckey in antag_rep_change)
+		//var/start = antag_rep[p_ckey]
```
Commented out code?

AndrewMontagne commented on code/controllers/subsystem/persistence.dm:
```diff
@@ -197,3 +215,48 @@ SUBSYSTEM_DEF(persistence)
+	for(var/p_ckey in antag_rep_change)
+		//var/start = antag_rep[p_ckey]
+		antag_rep[p_ckey] = max(0, min(antag_rep[p_ckey]+antag_rep_change[p_ckey], ANTAG_REP_MAXIMUM))
+		//WARNING("AR_DEBUG: [p_ckey]: Committed [antag_rep_change[p_ckey]] reputation, going from [start] to [antag_rep[p_ckey]]")
```
Commented out code?

AndrewMontagne commented on code/controllers/subsystem/persistence.dm:
```diff
@@ -197,3 +215,48 @@ SUBSYSTEM_DEF(persistence)
+	// people who died or left should not gain any reputation
+	// people who rolled antagonist still lose it
+	if(failed && SSpersistence.antag_rep_change[p_ckey] > 0)
+		//WARNING("AR_DEBUG: Zeroed [p_ckey]'s antag_rep_change due to failure")
```
Commented out code?

AndrewMontagne commented on code/controllers/subsystem/persistence.dm:
```diff
@@ -197,3 +215,48 @@ SUBSYSTEM_DEF(persistence)
+	//If they changed jobs to a higher rep, it'll still be the original job's rep so let's only worry about lower
+	if(current_job.antag_rep < original_job.antag_rep)
+		if(SSpersistence.antag_rep_change[p_ckey] > 0)
+			//WARNING("AR_DEBUG: Reduced [p_ckey]'s antag_rep_change due to higher rep job change")
```
Commented out code?

AndrewMontagne commented on code/game/gamemodes/game_mode.dm:
```diff
@@ -310,6 +310,59 @@
+		if(current <= antag_select)
+			var/subtract = min(p_rep + DEFAULT_ANTAG_TICKETS, MAX_TICKETS_PER_ROLL) - DEFAULT_ANTAG_TICKETS
+			SSpersistence.antag_rep_change[p_ckey] = -subtract
+			//WARNING("AR_DEBUG: Player [mind.key] won spending [subtract] tickets from starting value [SSpersistence.antag_rep[p_ckey]]")
```
Commented out code?

ike709 commented:
@AndrewMontagne aaaaaaaaaah okay I'll add a define for rep debugging

scrubmcnoob commented:
I feel that getting players into command at times can be troublesome, having incentive to play these roles will reward them.
Also as said earlier, which I think is the really good part, dying now has more weight to it. People will be less likely to easily throw their lives away. Hopefully this will encourage people to be more careful, and cautious when dealing with situations rather than charging straight in.

SamHPurp commented:
imo, if we're gonna merge this, it should also revert #625

ike709 commented:
I agree. I'll add that in later.

Tayyyyyyy commented:
I'm gonna copy and paste my analysis from tg for why this is a bad idea, especially with the values you have right now, and especially because Oracle is supposed to be an MRP server. You're rewarding mechanically important roles in a persistent way while punishing roles that are arguably just as important and add very valuable RP to any server that wants to be more than LRP.

> You cannot have negative points (i.e. you will not be discriminated against with negative personal odds)

With the way `antag_pick` is written right now, this isn't true at all. The current ticket system is a zero-sum game, where other people having a higher number of tickets than you means that you have a smaller chance of being picked. There's even an example in the comments of the code that illustrates this:

```
// The odds of being picked are simply (your_tickets / total_tickets). Suppose you have one player using fifty (50) extra tickets, and one who uses no extra:
//     Player A: 150 tickets
//     Player B: 100 tickets
//        Total: 250 tickets
//
// The odds become:
//     Player A: 150 / 250 = 0.6 = 60%
//     Player B: 100 / 250 = 0.4 = 40%
```
Assume you are player B and compare to just using normal random selection, where your chance of getting antag would've been 50%. It's now been lowered to 40%.

Just because everyone is getting tickets doesn't mean you won't be less likely to be chosen as antag. The number of tickets itself doesn't matter, it's the share of tickets you have.

For example, let's say there's a round with 10 people in it. Everyone spends 100 points. Nobody has actually been rewarded. They each have a 10% chance of getting antagonist.

The logic applies similarly to a round with 10 people where 9 people have spent 100 points but 1 person is an assistant main and has decided not to spend their points. Now, after some quick math, it turns out that the 9 people who each spent 100 points have a 10.53% chance of getting antag (a tiny boost), but that assistant main? They now only have a 5.26% chance of getting antag, rather than the 10% chance RNG would've given them. So the assistant main has been punished, and the other peeps have baaaaaarely been "rewarded".

To truly double the probability of someone getting antag, you have to increase the number of antags chosen based on the number of tickets. Probability always sums to one. If you don't increase the size of the pie, increasing your slice necessarily means decreasing someone else's.

To put it more simply, on average, people who play something like bartender often will get antagonist less often than before, while those who play captain/AI/etc will get it more often than before.

Tayyyyyyy commented:
With your numbers, the disparity is even worse than in my above analysis, which was pointing out issues with the tg implementation. At least in the tg implementation, only assistant was really getting heavily punished. In this version, detective and AI are 10 points, and captain is 20 points. Assuming it takes 200 points, like the tg implementation to reach the threshold where you get a higher chance of being picked, it takes 10 rounds for captain to reach this stage and 20 rounds for detective and AI. Meaning detective and AI players will be have lower antag chances for 19 rounds before they have a higher chance for antag, as opposed to captain. Assistant will take a whopping 65 rounds of lowered antag chances. This is very, very harsh. There's a reason tg basically neutered all the numbers on its version of the PR.

Tayyyyyyy commented:
Also, this version doesn't work. Make sure to take a look at tgstation/tgstation#36188

ike709 commented:
God damn it. I'll port that when it's merged.

ike709 commented:
Also, using your example with 9 people and the assistant main, that is indeed something I want to happen. I don't want an assistant main's chances to roll antag to be as good as a real job main's chances of rolling antag.

Tayyyyyyy commented:
My issue is that these claims are mathematically false:
> your chances of rolling antag increases until the odds are double

> Yeah it's pretty great. Just like normal SS13 except now sometimes you're more likely to be an antag.

The correct way to say this is that you will have lowered chances to roll antag, until you hit 200 points, at which point you may or may not have an increased probability to be antag. Your odds do NOT double; you just have double the chance of someone who isn't using their tickets. As I showed in my example, it depends on how many other people have also hit 200 points that round. The more people with 200 points that round, the less your chance of being antag will increase.

If you're trying to solve the problem of people not playing security or head roles, then that's a playerbase problem. Bribing people with antagery will only bring low quality play into those roles. Treating antagery as a currency just puts being antag on a pedestal further, worsening antag fishing and setting a bad precedent for the entire server. This PR is a mistake for an MRP server like Oracle, and arguably even for an LRP server like tg.

Also, that PR is just a fix to the antag picking process. This PR literally does nothing and disables itself if you don't implement that fix.

ike709 commented:
I mean I'll port the fix when the fix PR is merged.

ike709 commented:
Also while it's highly likely that I'm totally misunderstanding this code, I'm going to instead say that you are and refer to what MSO said to you on the tgstation pull request for this.

ike709 commented:
There's also the option in which this gets merged and if things don't pan out it can be reverted. I don't get why there's so much controversy over some PRs. You can't break the game forever with a single PR, a revert is just a click away if things don't go well.

SamHPurp commented:
@Tayyyyyyy thank you for sharing all the above, I completely agree with your assessment, and is how I feel about it too. Also the maths work is interesting, so thank you for sharing this.

Tayyyyyyy commented:
@ike709 MSO didn't understand what my post said. I didn't want to drag out the discussion any further since I don't play on tg and it seemed like it would get merged anyways. The problem with what MSO is saying is that there will almost always be people who are using their 200 points. This means that any time you aren't using your 200 points, you have a _decreased_ chance of being antag, due to your probability mass being taken away from you. There is no getting around that, it's just math. Also my 10 person example above assumes what MSO is assuming, that all players play at the same time and are in sync with each other.

@SamHPurp no problem, I've learned through doing terribly in probability related classes that probability is really hard and easy to mess up, so just trying to help avoid this here 😛

ike709 commented:
¯\\\_(ツ)_/¯ 

I don't have any math credits this year, so I have no desire to read into this enough to actually figure it out one way or another. I'm fond of the "let's just see what happens" approach.

Ziiro commented:
Let's look at the overall goals of this PR:

* Reward non-assistant play
* Reward smart/good play (Live until the end, don't suicide, don't get demoted)
* Reward playing roles with more responsibility

Even if the math is harsher than we expect and the exact math is misinterpreted (Thank you for the analysis, @Tayyyyyyy !), I think it still accomplishes these things.

ike709 commented:
woops

Tayyyyyyy commented:
This rewards only if:
- You are using your 200 points
- There are not many other people using their 200 points in the same round as you

This punishes otherwise. Even if you're a captain main, if you're not using your 200 points, you will have a lower antag chance compared to RNG because of probability mass taken from you. But captain main gets punished the least. Everyone else gets punished for not being a captain main. 

I don't see how this rewards responsibility at all. This punishes you unless you pick jobs you don't want to do or aren't qualified for, and barely rewards you.

Rewarding not dying is problematic for a mid RP server, too. Sometimes taking risks is good. Sometimes, dying can make for great story telling. Introducing meta-incentives to influence how the round goes is a _terrible_ idea for a MRP server.

This is not a case of the math being 'slightly off'. This is a case of completely misunderstanding the probability behind this as to think that this is "rewarding" people when, the vast majority of the time, you will be experiencing decreased chances of being antag. 

A chaplain main will take 39 rounds of reduced antag chances before they get their increased antag chance. A mime main will take 49 rounds of reduced antag chances. Add in dying or having to leave early and that's an absolutely massive amount of time for reduced antag chances.

You are getting very little benefit from this system and introducing the risk for a lot of server-breaking, community-harming, unintended consequences.

Ziiro commented:
>Rewarding not dying is problematic for a mid RP server, too. 

Strongly, strongly disagree. I think that people have no self-preservation and that ruins a bit of immersion. That's why they hunt down the dude with an armblade when all they've got is some soap and some cuffs.

Mathematically, what changes would you propose? I'm kind of curious.

Tayyyyyyy commented:
> I think that people have no self-preservation and that ruins a bit of immersion. That's why they hunt down the dude with an armblade when all they've got is some soap and some cuffs.

I strongly agree with this too. I just think this should be done through in game methods, not out of game methods. Remove roundstart cloning. Make cloning have a chance to swap out your soul. Limit clone chances to 3. There are so, so many legitimate ways to make death more meaningful than "hey, you won't have your antag chances reduced for 1 round out of the next 40 rounds you play".

As for the math part, you'd have to add probability mass to even it out. Let's say we're trying to pick one antagonists out of three players. Under the RNG model, I would have a 33% chance of getting picked.

Now let's say one of the other players has twice the tickets I do, and the rest of the players have the normal number of tickets. I and the other normal player have a 25% chance of getting picked, while the lucky player has a 50%. To bring the probability of getting picked back up to 33% (over the entire course of antag picking), we need to add the a random chance of picking another antag with some amoint of probability to restore the mass the lucky player stole from us. Unfortunately my brain is too fried right now to do the math for this solution.

Or, pick antags completely RNG the first run. Then, do a dummy run of the RNG again, where only 200 point players who didn't get picked in the first round are included (and dummy players are included to ensure that probabilities remain the same). This literally doubles their chances, and allows everyone else to have the same chance of getting antag as before. This solution, I think, is much easier even for my sleepy brain.

But basically you need to increase the number of antags picked. You can't make someone's slice of the pie bigger without making someone else's smaller, unless you increase the size of the pie.

ike709 commented:
Except I'm okay with making someone else's smaller.

ike709 commented:
The annoyance this PR is causing me will soon surpass how much I actually care about this PR.

avosirenfal commented:
@Tayyyyyyy Points are spent whenever you roll antag, you can just spend _up to_ 100 extra points (totaling 200) if you have them. It should be rare for someone to ever have and spend maximum points.

Tayyyyyyy commented:
Oh I see! Oops! But even if the points aren't spent all at once, and instead gradually spent, the effect is still there. A bartender main will consistently have a worse chance of getting antag than RNG dictates because they'll have less probability mass compared to all the other people who have way more. A system where you build up points to letting you participate in the 2nd round of antag selection is much more intuitive and less punishing.

In the end, the result is still that lower point scoring roles get antag less often than they did pre-PR, there's no way around that, even if you spend points at a trickle.

avosirenfal commented:
@Tayyyyyyy That's true, but that's the _point_. It was intended to discourage people from playing assistant solely to roll for antagonist, which it certainly does.

As for encouraging people to play more important roles, it may or may not do that depending on whether people care about being antagonist. But you have to ask yourself how many people "main" one and only one role forever, and whether those people _should_ be playing antagonist often if they do. If security, for instance, is harder to play, surely that's the kind of people you want in the most important role in the round (antag), right? That was my rationale, anyway.

Tayyyyyyy commented:
Thing is, it's really hard to tell which roles are going to see a net increase in antag chances and which will see a decrease, because with this system pretty much the only way to calculate this is to run simulations or observe the effect over time. It's really opaque and difficult to understand (at least in tg's version, the values were all clustered around the same things), which means people will make bad assumptions about what the effect of this system will be (like thinking this system only rewards people).

I wish I'd thought of the alternate 2-round antag selection idea earlier or I would've suggested it on your PR, but I honestly think it's a much more transparent, easy to understand way to approach this, even though I disagree with dangling antag carrots on principle.

Tayyyyyyy commented:
Players don't have to understand code, but the people adding stuff to the game should at least have an idea of what they're adding. If it's presented in the wrong way, players will just take their word for it. I mean, at that point, you may as well just merge in this version without your fix, and just use the placebo effect to encourage people to not play assistant.

There's a reason there's legislation and stuff to push for microtransaction loot box transparency or whatever. Not saying this is the same thing as child gambling, but this sort of metagame system that impacts the game in a big way should be transparent to players.

ike709 commented:
@Tayyyyyyy 

>implying I've _ever_ understood any of the code that I port

Fun fact: Despite popular belief, I have no idea how to code beyond cheesing the compiler. I just copy and paste code that sounds fun.

avosirenfal commented:
@AndrewMontagne Make sure https://github.com/tgstation/tgstation/pull/36188 comes with this too, it has a bugfix besides the value changes.

ike709 commented:
notice how the OP includes "DNM pending a fix port from TG."

also notice how our values are wildly different

avosirenfal commented:
@ike709 Yes... my point was just to make sure Andrew noticed that since he approved the PR already, I didn't want anyone to merge it without the fix. Also to make sure the values here weren't accidentally overwritten with the ones from the TG fix PR. No need to be snarky bruv.

AdamElTablawy commented:
>playing borg is worth less than playing atmos tech

Ziiro commented:
Atmos tech doesn't get emagged or subverted

SideCat commented:
I think we should give it a try, if it won't work out for people, we can always return to basic roll antag system.

SideCat commented:
I think we could give it 1 week try, if people will complain about it, then you could just remove it and bring back the good ol' antag roll system. Though, I would like to add that I think Shaft Miners should give atleast 12 or 14 points, since they're quite valuable and always taking their lifes at risk.

alexpkeaton commented:
As a frequent Captain, I would support halving the Cap rate from 20 to 10.  Here's what would happen: people will see Cap rate as highest rate to get antag, set Cap to High for the 20 rate when they don't roll antag, and be terrible in the job.
 
This would just force the shitters into lower head roles where their shittery is both subject to view by department members and demotion by the Cap.

Further, how will the code account for job name changes at HoP?  If a clown asks for a job name change to "Head of Clowning" will they still get the points at roundend or will they get 0?

Tayyyyyyy commented:
Debating numbers or testing this isn't very helpful without a robust logging system that tracks what players roll what and how often as well as how often they roll antagonist and "win" in this system. Your chances of winning antag depend on how many points every other player has rolling with you. A massive number of variables goes into that and like @avosirenfal said you cannot predict what roles this system will incentivize or disincentivize beyond the placebo effect. In one round having 140 points could give you a boost over everyone else. In another round having 140 points could _lower_ your chances of getting antag.

If you want to get into debating what values to give each role, the system needs to be predictable and clear.

Ziiro commented:
Alright, so let's standardize.

Base: +3 Points for surviving a shift and keeping your job

---

Then we take the department:
* +1 For Science // You don't need to be rewarded for playing science. Science is its own reward.
* +2 For Cargo
* +2 For Service
* +3 For Engineering
* +4 For Medical
* +5 For silicons, as they can be subverted and can be a stressful role.
* +10 For Security // We'll catch people who play sec for points and are shitcurity and job ban them

---

Misc additions:
* +3 For Command-related
* +5 For mindshielded roles  // Can't be converted, more likely to be killed

---

Head of Security - 21
Warden - 18
Detective - 18
Security Officer - 18
Brig Physician - 12
Captain - 11
Internal Affairs Agent - 11 // No department, really.
Blueshield - 11
AI - 11
Chief Medical Officer - 10
Chief Engineer - 9
Cyborg - 8
Quartermaster - 8
Head of Personnel - 7
Research Director - 7
Chemist - 7
Medical Doctor - 7
Virologist - 7
Paramedic - 7
Atmospheric Technician - 6
Station Engineer - 6
Botanist - 5
Cook - 5
Janitor - 5
Shaft Miner - 5
Bartender - 5
Cargo Technician - 5
Chaplain - 5
Clown - 5
Curator - 5
Mime - 5
Geneticist - 4 // More Science than Medical, if we're honest.
Roboticist - 4
Scientist - 4
Assistant - 3

`>scientist is lower than clown`
🤡

Tayyyyyyy commented:
Standardizing something that is arbitrary and meaningless doesn't make it any less arbitrary and meaningless, besides making it more palatable to the uninformed player. Here's what I mean. Let's look at a few examples. 

You have 130 points from playing CMO three rounds in a row. In scenario one, there are 9 other players in the round with you, and they're all first time players and so don't have any points (meaning they each have 100 tickets). That gives you a 12% chance of being antag, a moderate boost from RNG 10%.

Now let's say the other 9 players are captain mains who have been unlucky and are all maxed out at 200 points. Now your probability of being antag is 6%, which is almost half that you would get from RNG.

Or, let's say there's 19 other players, and they average out at 150 points each. Your probability of getting antag is now about 4%, compared to 5% RNG.

As you can see, the chance you have of being antag and the number of points you have vary _wildly_ depending on who's rolling with you. This is not a true reward/punishment system because we have _zero idea of what is being rewarded or punished at all_. This PR almost entirely relies on the placebo effect to achieve its goal.

Compare this to the second round antag system. It's straightforward and easy to balance. Captain is worth 20 points? It takes 10 rounds for them to reach the second round of antag selection. Is captain being rewarded too much? Decrease their point value to 10 to halve the amount of reward.

Or maybe too many people are reaching the second round of antag selection? Up the point threshold. And as a bonus, you're not punishing anyone!

ike709 commented:
TG is taking too long to merge the fix PR so I went ahead and ported it anyways.

AdamElTablawy commented:
Yeah honestly this is kind of a terrible idea now that I think about it.

SmokingKilz commented:
Right so, after a bunch of discussions on this, here's how we're going to deal with this. We're going to shelf this and wait for TG to actually test it out themselves, then once we can get some more specific info from them, we'll re-visit it. 

Also, for reference, yes TG has merged the PR but it's not active in their configs.

Derpyhsi commented:
So its been a month. Has there been any progress on this PR?

ike709 commented:
Does that answer your question?

FlattestGuitar commented:
While we likely won't be using this anytime soon, do you mind if I later reopen the PR and merge it?

ike709 commented:
sure go nuts
